lib/modules/apostrophe-areas/public/js/editor.js

```diff
@@ -644,25 +645,40 @@ apos.define('apostrophe-areas-editor', {
           'top': '',
           'width': ''
         });
-        $(event.target).after($item);
+        var $oldWidget = $item.find('[data-apos-widget]');
+        var oldEditor = self;
+        var $newArea = $(event.target).closest('[data-apos-area]');
+        var newEditor = $newArea.data('editor');
+        $oldWidget.data('areaEditor', newEditor);
         self.disableDroppables();
-        self.reRenderWidget($item);
-        self.changeOwners($item);
+        oldEditor.startAutosavingThen(function() {
```

Comment:
Is it worth checking here if `oldEditor` and `newEditor` are different before doing extra work?

Reply:
It'd be a matter of performance hit, which might only be noticeable if a really beefy and nesting widget.

Reply:
We have always rerendered in this situation, so I'm going to leave it alone for this release rather than reopen the kettle of worms.